spacy_llm/tasks/lemma/task.py

```diff
@@ -0,0 +1,96 @@
+from typing import Any, Callable, Dict, Generic, Iterable, List, Optional, Type
+
+import jinja2
+from spacy.language import Language
+from spacy.scorer import Scorer
+from spacy.tokens import Doc
+from spacy.training import Example
+
+from ...ty import FewshotExample, TaskResponseParser
+from ..templates import read_template
+from ..util import ExampleType, SerializableTask
+
+DEFAULT_LEMMA_TEMPLATE_V1 = read_template("lemma")
+
+
+class LemmaTask(SerializableTask[ExampleType], Generic[ExampleType]):
+    def __init__(
+        self,
+        template: str = DEFAULT_LEMMA_TEMPLATE_V1,
+        parse_responses: Optional[TaskResponseParser] = None,
+        fewshot_example_type: Optional[Type[FewshotExample]] = None,
+        examples: Optional[List[ExampleType]] = None,
+    ):
+        """Default lemmatization task.
+
+        template (str): Prompt template passed to the model.
+        parse_responses (Optional[TaskResponseParser]): Callable for parsing LLM responses for this task.
+        fewshot_example_type (Optional[Type[FewshotExample]]): Type to use for fewshot examples.
+        examples (Optional[Callable[[], Iterable[Any]]]): Optional callable that
+            reads a file containing task examples for few-shot learning. If None is
+            passed, then zero-shot learning will be used.
+        """
+        self._template = template
+        self._parse_responses = parse_responses
+        self._prompt_examples = examples or []
+        self._fewshot_example_type = fewshot_example_type
+
+        assert callable(self._parse_responses)
+        assert self._fewshot_example_type is not None
```

Comment:
Let's discuss this a bit more internally - it feels weird to type these as `Optional` and then immediately `assert` that they're not `None`. It's all related to the serialization & needing `None` defaults.

Reply:
Fixed in https://github.com/explosion/spacy-llm/pull/242/commits/46fc5cfc0f498ef54a68a3910e5c0653cbcf20dd.